### PR TITLE
[windows] SourceKit tests with VFS fixes

### PR DESCRIPTION
**Pull request by drodriguez**
The VFS tests were using Unix absolute paths, which does not play well
when Windows see them as relative to the current drive letter.

By using the temporal directory, both Windows and Unix can use the same
paths and avoid the problem.

Additionally, a couple of inputs have to be transformed into the native
path format, because sourcekitd-test compares the inputs as strings, and
they need to match exactly. So the source file and the name of the VFS
entries are transformed into native using the helper from LLVM support.

### CODE REVIEW

**Comment by drodriguez** [comment]
@swift-ci please clean test Windows platform

**Comment by drodriguez** [comment]
```
    Swift(windows-x86_64) :: SourceKit/Sema/injected_vfs_sourcetext.swift
    Swift(windows-x86_64) :: SourceKit/Sema/injected_vfs.swift
```

Fail for me also locally with the same `LLVM ERROR: Never got notification for semantic info`. I don't understand exactly why.

```
Swift(windows-x86_64) :: SourceKit/InterfaceGen/gen_swift_module.swift
```

Regression.

```
    Swift(windows-x86_64) :: Driver/sanitize_recover.swift
```

Unrelated. Broken in master since https://ci-external.swift.org/job/oss-swift-windows-x86_64/1953/. Probably introduced in #28126

**Comment by gmittert** [comment]
cc @akyrtzi

File: test/SourceKit/CursorInfo/injected_vfs.swift [hunk 1]
```diff
@@ -10,7 +10,7 @@ func foo(
 
 // CHECK-CMODULE: key.kind: source.lang.swift.ref.struct
 // CHECK-CMODULE: key.name: "StructDefinedInCModule"
-// CHECK-CMODULE: key.filepath: "/CModule/CModule.h"
+// CHECK-CMODULE: key.filepath: "{{.*}}/CModule{{/|\\\\}}CModule.h"
```
Comment (gmittert) [inline test/SourceKit/CursorInfo/injected_vfs.swift]: Why is one slash here always a forward slash, but the other can be a backslash?

**Comment by gmittert** [comment]
@swift-ci please smoke test

File: test/SourceKit/CursorInfo/injected_vfs.swift (hunk 1 above)
Comment (drodriguez) [reply test/SourceKit/CursorInfo/injected_vfs.swift]: Welcome to Windows™️! :D

(Yes, one of them was a `/` even on Windows, so I didn't replace it. Adding the `{{.*}}` accounts for the new path relative to `%t`, while the `{{/|\\\\}}` accounts for an escaped `\` in Windows.

**Comment by drodriguez** [comment]
The `gen_swift_interface.swift` should be fixed with the latest changes.

@swift-ci please clean test Windows platform

**Comment by drodriguez** [comment]
@swift-ci please clean test Windows platform

**Comment by drodriguez** [comment]
```
    Swift(windows-x86_64) :: SourceKit/Sema/injected_vfs_sourcetext.swift
    Swift(windows-x86_64) :: SourceKit/Sema/injected_vfs.swift
```

Still happening (expected)

```
   Swift(windows-x86_64) :: SourceKit/InterfaceGen/gen_stdlib.swift
```

Regression :(

```
    Swift(windows-x86_64) :: ClangImporter/pcm-emit-and-import.swift
```

Failing in master. Sigh.

**Comment by drodriguez** [comment]
Changes to make gen_stdlib.swift pass. Thanks to @gmittert for the solution.

@swift-ci please test Windows platform

**Comment by benlangmuir** [comment]
@swift-ci please test

**Comment by benlangmuir** [comment]
@swift-ci please test

**Review by benlangmuir — APPROVED** [review]

**Comment by drodriguez** [comment]
Thanks again to @gmittert, who found the last point in the code that was creating problems.

@swift-ci please test Windows platform
@swift-ci please smoke test

**Comment by drodriguez** [comment]
@swift-ci please test Windows platform

**Comment by drodriguez** [comment]
https://ci-external.swift.org/view/Pull%20Request/job/swift-PR-windows/177/ failed because of #28280, which is testing and I hope to land soon. Sigh.

**Comment by drodriguez** [comment]
@swift-ci please test Windows platform

**Review by gmittert — APPROVED** [review]

**Comment by drodriguez** [comment]
Removed the explicit slashes in the unit test, which should make it work in both OS.

@swift-ci please smoke test

**Comment by drodriguez** [comment]
@swift-ci please test Windows platform

**Comment by gmittert** [comment]
@swift-ci please test Windows platform

**Comment by gmittert** [comment]
Thanks for doing all that tedious path fiddling!

**Comment by drodriguez** [comment]
Thanks to you for all the help!

**Pull request merged**